controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -169,6 +170,13 @@ func (r *OCSInitializationReconciler) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	clusterClaim := &ocsClusterClaim{}
+	reconcileResult, err := clusterClaim.ensureCreated(r, instance)
+	if err != nil {
+		r.Log.Error(err, "Failed to ensure odf-info namespacedname ClusterClaim")
+		return reconcileResult, err
+	}
+
```

Comment:
This is not a full solution, as we need to trigger an event whenever an external change is forced on the claim (like someone deleting it). This means we need to add a watch as part of `SetupWithManager` but you cannot watch a type if the CRD does not exist, this means we need code to conditionally watch and another watch for the creation of the CRD itself to issue a pod restart (via panic) for cases where the CRD is introduced into the cluster after the operator is already running.

This use case, of the need to watch a CR of a CRD that might or might not exist on a cluster, becomes more and more common. We might want to consider implementation as part of utils to be used as a common implementation.

Reply:
> We might want to consider implementation as part of utils to be used as a common implementation.

I've already added it [here](https://github.com/red-hat-storage/ocs-client-operator/pull/168). I just need to convert it to a util function.